ballerina/azure_ai_search_knowledgebase.bal

```diff
@@ -107,17 +114,20 @@ public distinct isolated class AiSearchKnowledgeBase {
     #                                  This configuration is only required when the `index` parameter is 
     #                                  provided as an `search:SearchIndex`
     # + indexClientConnectionConfig - Connection configuration for the Azure AI index client.
+    # + semanticConfigurationDetails - Optional semantic configuration details for semantic search.
     # + return - An instance of `AiSearchKnowledgeBase` or an `ai:Error` if initialization fails
     public isolated function init(string serviceUrl, string apiKey, 
-            string|search:SearchIndex index, ai:EmbeddingProvider embeddingModel, 
+            string|search:SearchIndex index, ai:EmbeddingProvider? embeddingModel = (), 
             ai:Chunker|ai:AUTO|ai:DISABLE chunker = ai:AUTO, boolean verbose = false, 
             string apiVersion = AI_AZURE_KNOWLEDGE_BASE_API_VERSION, string contentFieldName = CONTENT_FIELD_NAME, 
             search:ConnectionConfig searchClientConnectionConfig = {},
-            index:ConnectionConfig indexClientConnectionConfig = {}) returns ai:Error? {
+            index:ConnectionConfig indexClientConnectionConfig = {},
+            SemanticConfigurationDetails? semanticConfigurationDetails = ()) returns ai:Error? {
```

Comment:
if the SemanticConfigurationDetails just have the name can't we introduce a string value instead of record?
`semanticConfigurationName` ?